Description: Always append the index of the font within the font file to the font file name.
 It is ignored (and always "0" anyway) by otf_load() for .ttf files but crucial for .ttc files.
Author: Fabian Greffrath <fabian+debian@greffrath.com>

--- cups-filters-1.0.4.orig/filter/texttopdf.c
+++ cups-filters-1.0.4/filter/texttopdf.c
@@ -57,7 +57,7 @@ EMB_PARAMS *font_load(const char *datadi
     FcPattern *pattern;
     FcFontSet *candidates;
     FcChar8   *fontformat, *fontname = NULL;
-    int i, spacing;
+    int i, index, spacing;
 
     /* Remove extension from the passed file name
        to turn it into a minimal "pattern" in the fontconfig sense */
@@ -77,10 +77,11 @@ EMB_PARAMS *font_load(const char *datadi
        find the first one that is both in TrueType format and monospaced */
     for (i = 0; i < candidates->nfont; i++) {
       FcPatternGetString  (candidates->fonts[i], FC_FONTFORMAT, 0, &fontformat);
+      FcPatternGetInteger (candidates->fonts[i], FC_INDEX,      0, &index);
       FcPatternGetInteger (candidates->fonts[i], FC_SPACING,    0, &spacing);
 
       if ((strcmp(fontformat, "TrueType") == 0) && (spacing == FC_MONO)) {
-        fontname = FcPatternFormat (candidates->fonts[i], "%{file|cescape}");
+        fontname = FcPatternFormat (candidates->fonts[i], "%{file|cescape}/%{index}");
         break;
       }
     }
